shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionFeature.kt

```diff
@@ -76,11 +76,25 @@ interface StepCompletionFeature {
             data class Error(val errorMessage: String) : FetchNextRecommendedStepResult
         }
 
+        /**
+         * Show problem of day solve modal
+         */
+        data class ShowProblemOfDaySolvedModal(val earnedGemsText: String) : Message
+        object ProblemOfDaySolvedModalGoBackClicked : Message
+
+        /**
+         * Daily study reminders
+         */
+        object RequestDailyStudyRemindersPermission : Message
+        data class RequestDailyStudyRemindersPermissionResult(val isGranted: Boolean) : Message
```

Comment:
Action should be called like Do something. In this case I suggest to call it like EnableDailyStudyReminder and separate the implementation in the ActionDispatcher into 2 different actions, which is called by reducer.